### PR TITLE
WIP: network panel support for DevTools

### PR DESCRIPTION
**Pull request by lucacasonato**
Let's get `fetch` and `WebSocket` hooked up to the "Network" panel in Chrome DevTools.

### CODE REVIEW

File: runtime/worker.rs [hunk 1]
```diff
@@ -158,13 +158,16 @@ impl MainWorker {
...
+      let dev_tools_agent =
+        js_runtime.op_state().borrow().dev_tools_agent.clone();
       let inspector = js_runtime.inspector();
       let session_sender = inspector.get_session_sender();
       let deregister_rx = inspector.add_deregister_handler();
       server.register_inspector(
         session_sender,
         deregister_rx,
         main_module.to_string(),
+        dev_tools_agent,
```
Comment (caspervonb) [inline runtime/worker.rs:170]: So unless I missed something we end up the minor caveat here that local sessions won't support the same extended subset as remote sessions get?

Comment (bartlomieju) [reply runtime/worker.rs:170]: This is just a prototype that will be rewritten. I guess you're still referring to the ability to connect local REPL session to the Devtools?

Comment (caspervonb) [reply runtime/worker.rs:170]: > This is just a prototype that will be rewritten. I guess you're still referring to the ability to connect local REPL session to the Devtools?

Yeah it's low hanging fruit but hypothetically we'd want to expose the local inspector as as a cli binding ala node down the line.

**Comment by bartlomieju** [comment]
Stale, closing for now, I'll try to revisit this feature in Q1 2022

**Pull request closed without merging**